src/ucp/core/ucp_worker.c

```diff
@@ -2532,9 +2534,9 @@ void ucp_worker_destroy(ucp_worker_h worker)
     uct_worker_progress_unregister_safe(worker->uct, &worker->keepalive.cb_id);
     ucp_worker_destroy_eps(worker, &worker->all_eps, "all");
     ucp_worker_destroy_eps(worker, &worker->internal_eps, "internal");
-    ucp_worker_remove_am_handlers(worker);
     ucp_am_cleanup(worker);
     ucp_worker_discard_uct_ep_cleanup(worker);
+    ucp_worker_remove_am_handlers(worker);
```

Comment:
why changed the order here?

Reply:
see https://github.com/openucx/ucx/pull/7819#discussion_r788444753

Reply:
can you pls add code comment (before this line) with short explanation?

Reply:
done.